src/components/UserAuthentication/AutoLogout.tsx

```diff
@@ -33,22 +35,42 @@ function AutoLogout(props: Props): React.ReactElement {
     props.setAutoLogout(true);
   };
 
-  const warnUserIdle = (): void => {
+  const resetIdleTimer = (): void => {
+    setSecondsIdle(0);
+    clearTimeout(timerRef.current);
+    timerRef.current = setTimeout(handleAutoLogout, timeFromWarnToLogout);
+  };
+
+  const onIdle = (): void => {
     setShowModal(true);
-    // start the logout timer
-    const timeout = setTimeout(handleAutoLogout, timeFromWarnToLogout);
-    setLogoutTimeout(timeout);
+    resetIdleTimer();
   };
 
-  // closing idle modal warning
-  const handleClose = (): void => {
-    // reset the logout timer
-    clearTimeout(logoutTimeout);
+  const onClose = (): void => {
     setShowModal(false);
-    // reset the warn timer
-    idleTimerWarn.reset();
+    resetIdleTimer();
   };
 
+  useEffect(() => {
+    window.addEventListener('mousemove', resetIdleTimer);
+    window.addEventListener('mousedown', resetIdleTimer);
+    window.addEventListener('keypress', resetIdleTimer);
+
+    return () => {
+      window.removeEventListener('mousemove', resetIdleTimer);
```

Comment:
Just wondering why are we removing the event listener? shouldn't we want to just reset the idle timer?